gapic/schema/wrappers.py

```diff
@@ -713,7 +713,7 @@ def with_context(self, *,
         references.
         """
         visited_messages = visited_messages or set()
-        visited_messages.add(self)
+        visited_messages = visited_messages | {self}
```

Comment:
@parthea How is adding an element `b` to a set `A` different than updating `A` to be the union of `A` and `{b}`?

Reply:
Great question. I don't have the answer. Issue https://github.com/googleapis/gapic-generator-python/issues/1794 is still open to try to create a minimal reproduction of the issue to dig deeper.